src/compiler/checker.ts

```diff
@@ -2172,7 +2175,7 @@ namespace ts {
             for (const decl of symbol.declarations) {
                 if (decl.kind === SyntaxKind.TypeParameter) {
                     const parent = isJSDocTemplateTag(decl.parent) ? getJSDocHost(decl.parent) : decl.parent;
-                    if (parent === container) {
+                    if (parent && parent === container) {

```

Comment:
container can never be `=== undefined`, so this additional check isn't needed.

Reply:
@sandersn done

Reply:
> you worked around this in your latest commit, right?

Yes, I moved logic to `getJSDocHost`